### ISSUE TITLE
x/review: mail must update the reviewer list even if there are no staged changes

### ISSUE THREAD

**Issue by rakyll** [golang/go#11192]
mail rejects to update the reviewer list if there are no staged changes. This behaviour is breaking a common use-case where the contributor creates a CL with no reviewers, review his/her changes on Gerrit and add reviewers. Allow mail subcommand to update the reviewers list even if there are no changes.

**Comment by adg** [comment]
Is it this behavior that you're referring to?

```
$ git mail 
remote: Processing changes: new: 1, done    
remote: 
remote: New Changes:
remote:   https://go-review.googlesource.com/11095 test: test commit, please ignore
remote: 
$ git mail  -r bradfitz
remote: Processing changes: refs: 1, done    
To https://go.googlesource.com/build
 ! [remote rejected] HEAD -> refs/for/master%r=bradfitz@golang.org (no new changes)
error: failed to push some refs to 'https://go.googlesource.com/build'
(running: git push -q origin HEAD:refs/for/master%r=bradfitz@golang.org)
git-codereview: exit status 1
```

This is the Gerrit server rejecting push, not the `git-codereview` tool.

I agree this workflow should work. I can see two ways of making this happen:

1. Fix it on Gerrit's side (paging @spearce, or I guess we could [file an issue](https://code.google.com/p/gerrit/issues/entry)),
2. Have the `git-codereview` tool do a `git diff HEAD $BRANCH.mailed` and—if there are no changes—use the Gerrit API to adjust the reviewers list instead of doing a `git push`.

I'd prefer to do the former rather than the latter. If @spearce indicates that this can happen on the Gerrit side, then I'll close this issue. Otherwise we should think about working around it on the `git-codereview` side.

**Comment by rakyll** [comment]
Yes, it's the behaviour above I was referring to. I assume the latter solution is easy to implement and it must relatively harder to get this done at the Gerrit side not to break the current behavioral compatibility.
